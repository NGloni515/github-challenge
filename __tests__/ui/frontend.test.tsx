import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import Home from '../../pages'

describe('Body', () => {
    beforeEach(()=>{
      global.fetch = jest.fn(()=>Promise.resolve({
        json: ()=>Promise.resolve({})
      })) as jest.Mock;
    });
  
    it('Match snapshot Home', async() => {
      const {asFragment} = render(<Home/>)
      expect(asFragment()).toMatchSnapshot()
    })

    it('renders a Table', async() => {
      const {getByTestId} = render(<Home/>)
      const table = getByTestId("commits-table")
      expect(table).toBeInTheDocument()
    })
  
    it('renders a Select', async() => {
      const {getByTestId} = render(<Home/>)
      const select = getByTestId("branches-select")
      expect(select).toBeInTheDocument()
    })
})