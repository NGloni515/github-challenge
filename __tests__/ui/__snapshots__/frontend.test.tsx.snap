// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Match snapshot Home 1`] = `
<DocumentFragment>
  <div
    class="h-screen bg-white"
  >
    <div
      class="pt-6"
    >
      <div
        class="md:grid md:grid-cols-4 md:gap-1"
      >
        <nav
          aria-label="Breadcrumb"
        >
          <ol
            class="mx-auto flex max-w-2xl items-center space-x-2 px-4 sm:px-6 lg:max-w-7xl lg:px-8"
            role="list"
          >
            <li>
              <div
                class="flex items-center"
              >
                <a
                  class="mr-2 text-sm font-medium text-gray-900"
                  href=""
                  target="_blank"
                />
                <svg
                  aria-hidden="true"
                  class="h-5 w-4 text-gray-300"
                  fill="currentColor"
                  height="20"
                  viewBox="0 0 16 20"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.697 4.34L8.98 16.532h1.327L7.025 4.341H5.697z"
                  />
                </svg>
              </div>
            </li>
            <li
              class="text-sm"
            >
              <a
                aria-current="page"
                class="font-medium text-gray-500 hover:text-gray-600"
                target="_blank"
              />
            </li>
          </ol>
        </nav>
        <div
          class="w-56"
        >
          <div
            class="relative mt-1"
            data-testid="branches-select"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="relative w-full cursor-default rounded-md border border-gray-300 bg-white py-2 pl-3 pr-10 text-left shadow-sm focus:border-indigo-500 focus:outline-none focus:ring-1 focus:ring-indigo-500 sm:text-sm"
              data-headlessui-state=""
              id="headlessui-listbox-button-:r0:"
              type="button"
            >
              <span
                class="flex items-center"
              >
                <svg
                  aria-hidden="true"
                  class="octicon octicon-git-branch"
                  data-view-component="true"
                  height="16"
                  version="1.1"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <path
                    d="M11.75 2.5a.75.75 0 100 1.5.75.75 0 000-1.5zm-2.25.75a2.25 2.25 0 113 2.122V6A2.5 2.5 0 0110 8.5H6a1 1 0 00-1 1v1.128a2.251 2.251 0 11-1.5 0V5.372a2.25 2.25 0 111.5 0v1.836A2.492 2.492 0 016 7h4a1 1 0 001-1v-.628A2.25 2.25 0 019.5 3.25zM4.25 12a.75.75 0 100 1.5.75.75 0 000-1.5zM3.5 3.25a.75.75 0 111.5 0 .75.75 0 01-1.5 0z"
                    fill-rule="evenodd"
                  />
                </svg>
                <span
                  class="ml-3 block truncate"
                >
                  main
                </span>
              </span>
              <span
                class="pointer-events-none absolute inset-y-0 right-0 ml-3 flex items-center pr-2"
              >
                <svg
                  aria-hidden="true"
                  class="h-5 w-5 text-gray-400"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M10 3a.75.75 0 01.55.24l3.25 3.5a.75.75 0 11-1.1 1.02L10 4.852 7.3 7.76a.75.75 0 01-1.1-1.02l3.25-3.5A.75.75 0 0110 3zm-3.76 9.2a.75.75 0 011.06.04l2.7 2.908 2.7-2.908a.75.75 0 111.1 1.02l-3.25 3.5a.75.75 0 01-1.1 0l-3.25-3.5a.75.75 0 01.04-1.06z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="p-10"
      >
        <div
          class="bg-gray overflow-auto rounded-lg shadow"
          data-testid="commits-table"
        >
          <table
            class="w-full"
          >
            <thead
              class="bg-gray-50 border-b-2 border-gray-200"
            >
              <tr>
                <th
                  class="p-3 text-sm font-semibold tracking-wide text-left"
                >
                  Comment 
                </th>
                <th
                  class="w-60 p-3 text-sm font-semibold tracking-wide text-left"
                >
                  Authored
                </th>
                <th
                  class="w-60 p-3 text-sm font-semibold tracking-wide text-left"
                >
                  Commited By
                </th>
                <th
                  class="w-32 p-3 text-sm font-semibold tracking-wide text-left"
                >
                  Date
                </th>
                <th
                  class="w-32 p-3 text-sm font-semibold tracking-wide text-left"
                >
                  URL
                </th>
                <th
                  class="w-32 p-3 text-sm font-semibold tracking-wide text-left"
                >
                  Hash
                </th>
              </tr>
            </thead>
            <tbody
              class="divide-y divide-gray-100"
            />
          </table>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
